libs/clients/assets/src/lib/properties-api.provider.ts

```diff
@@ -0,0 +1,48 @@
+import { Provider } from '@nestjs/common/interfaces/modules/provider.interface'
+import nodeFetch, { Request } from 'node-fetch'
+
+import { createEnhancedFetch } from '@island.is/clients/middlewares'
+import {
+  ConfigType,
+  LazyDuringDevScope,
+  XRoadConfig,
+} from '@island.is/nest/config'
+
+import { Configuration, FasteignirApi } from '../../gen/fetch'
+import { AssetsClientConfig } from './assets.config'
+
+export const PropertiesApiProvider: Provider<FasteignirApi> = {
+  provide: FasteignirApi,
+  scope: LazyDuringDevScope,
+  useFactory: (
+    xroadConfig: ConfigType<typeof XRoadConfig>,
+    config: ConfigType<typeof AssetsClientConfig>,
+  ) =>
+    new FasteignirApi(
+      new Configuration({
+        fetchApi: createEnhancedFetch({
+          name: 'clients-assets',
+          ...config.fetch,
+          autoAuth: {
+            mode: 'tokenExchange',
+            ...config.auth,
+          },
+          fetch: (url, init) => {
+            // The Properties API expects two different authorization headers for some reason.
```

Comment:
@hrefnalind Þjóðskrá is planning to ask us to call their webservice with two tokens:

Authorization: Bearer <innskra.island.is access token>
Authorization-Identity: Bearer <Þjóðskrá IDS access token>

It would be good to chat with them and dig into their concerns and see if we can convince them against this path. As I understand it, this PR already addresses the issue they describe since our public tokens won't have access to their scope/service anymore.